### PR TITLE
feat: [STO-6209]: Add Redis registration

### PR DESCRIPTION
**Pull request by r0610205**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50775)

### CODE REVIEW

**Comment by r0610205** [comment]
trigger uts

**Comment by r0610205** [comment]
trigger uts

File: 315-sto-manager/app/src/main/java/io/harness/app/STOManagerApplication.java [hunk 1]
```diff
@@ -504,6 +509,7 @@ private void registerWaitEnginePublishers(Injector injector) {
         injector.getInstance(NotifyQueuePublisherRegister.class);
     notifyQueuePublisherRegister.register(
         NG_ORCHESTRATION, payload -> publisher.send(singletonList(NG_ORCHESTRATION), payload));
+    notifyQueuePublisherRegister.register("sto_orchestration", injector.getInstance(STONotifyEventPublisher.class));
```
Comment (chris-harness) [inline 315-sto-manager/app/src/main/java/io/harness/app/STOManagerApplication.java:512]: Should `"sto_orchestration"` be moved into a named const?

**Review by GabeAboy — APPROVED** [review]

File: 315-sto-manager/app/src/main/java/io/harness/sto/execution/STONotifyEventConsumerRedis.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import javax.cache.Cache;
+
+@Singleton
+@OwnedBy(HarnessTeam.STO)
+public final class STONotifyEventConsumerRedis extends PmsAbstractRedisConsumer<STONotifyEventMessageListener> {
+  @Inject
+  public STONotifyEventConsumerRedis(@Named(STO_ORCHESTRATION_NOTIFY_EVENT) Consumer redisConsumer,
+      STONotifyEventMessageListener messageListener, @Named("stoEventsCache") Cache<String, Integer> eventsCache,
```
Comment (chris-harness) [inline 315-sto-manager/app/src/main/java/io/harness/sto/execution/STONotifyEventConsumerRedis.java:28]: const?

File: 332-ci-manager/app/src/main/java/io/harness/ci/CIManagerServiceModule.java [hunk 3]
```diff
@@ -441,13 +439,14 @@ private void registerEventListeners() {
...
-              CI_MANAGER.getServiceId(), redisConfig.getEnvNamespace()));
+          .annotatedWith(Names.named(this.configurationOverride.getOrchestrationEvent()))
+          .toInstance(GitAwareRedisProducer.of(this.configurationOverride.getOrchestrationEvent(), redissonClient, 5000,
+              this.configurationOverride.getServiceHeader().getServiceId(), redisConfig.getEnvNamespace()));
 
       bind(Consumer.class)
-          .annotatedWith(Names.named(CI_ORCHESTRATION_NOTIFY_EVENT))
-          .toInstance(RedisConsumer.of(CI_ORCHESTRATION_NOTIFY_EVENT, CI_MANAGER.getServiceId(), redissonClient,
+          .annotatedWith(Names.named(this.configurationOverride.getOrchestrationEvent()))
+          .toInstance(RedisConsumer.of(this.configurationOverride.getOrchestrationEvent(),
```
Comment (chris-harness) [inline 332-ci-manager/app/src/main/java/io/harness/ci/CIManagerServiceModule.java:448]: nit: repeated calls to `getOrchestrationEvent()` could be replaced with one call and a const

**Review by chris-harness — DISMISSED** [review]
Looks reasonable, with a few minor nitpicks

**Comment by r0610205** [comment]
trigger uts

**Review by chris-harness — APPROVED** [review]

**Review by atmollo-harness — APPROVED** [review]

**Review by harshjain12 — APPROVED** [review]

**Pull request merged**